src/app/(BottomTabNavigation)/AllCases/QRCodeScanner/index.tsx

```diff
@@ -20,9 +20,21 @@ enum permissions {
 function QRCodeScannerScreen() {
   const [hasPermission, setHasPermission] = useState(permissions.UNDETERMINED);
   const [scanned, setScanned] = useState<boolean>(false);
+  const [validIds, setValidIds] = useState<CaseUid[]>([]);
+  const [userIds, setUserIds] = useState<CaseUid[]>([]);
   const [toast, setToast] = useState<string>();
   const navigation = useNavigation();
 
+  useEffect(() => {
+    const getAllCasesAndValidCases = async () => {
+      const allCases = await getAllCaseIds();
+      setValidIds(allCases);
+      const userCases = await getCaseIdsFromUserId('NO_ID');
+      setUserIds(userCases);
+    };
+    getAllCasesAndValidCases();
+  }, []);
+
   useEffect(() => {
     navigation.addListener('blur', async () => {
```

Comment:
In these events, let's also add `setToast('')` so that if a user scans an invalid/duplicate case, sees the error, scans a valid case, and clicks cancel, their QR Code Scanner screen won't display an error. Let me know if that doesn't make sense!